packages/flutter_tools/bin/xcode_debug.js

```diff
@@ -0,0 +1,519 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview OSA Script to interact with Xcode. Functionality includes
+ * checking if a given project is open in Xcode, starting a debug session for
+ * a given project, and stopping a debug session for a given project.
+ */
+
+'use strict';
+
+function run(args_array = []) {
+  let args;
+  try {
+    args = new CommandArguments(args_array);
+  } catch (e) {
+    return new RunJsonResponse(false, `Failed to parse arguments: ${e}`);
+  }
+
+  const xcodeResult = getXcode(args);
+  if (xcodeResult.error != null) {
+    return new RunJsonResponse(false, xcodeResult.error).stringify();
+  }
+  const xcode = xcodeResult.result;
+
+  if (args.command === 'project-opened') {
```

Comment:
`project-opened` doesn't sound like a command. What does it do? What if there are multiple projects?

Reply:
Yeah I wasn't sure what to name it. It checks if the project is opened. If you check the comment of the function it calls, it explains it. It gets all workspaces opened in Xcode and matches on the projectPath/workspacePath. Unless you're doing something weird, you shouldn't have multiple open workspaces open that have the same path, but if you do, it'll return the first it finds.

Perhaps a better name is `check-project-opened`?